### ISSUE TITLE
configure fails with busybox mktemp

### ISSUE THREAD

**Issue by ncopa** [openzfs/zfs#11274]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Alpine Linux
Distribution Version	| 3.13.0_alpha20200917 (aka 'edge')
Linux Kernel	|  5.4.27-0-lts
Architecture	| aarch64 and x86_64 (at least)
ZFS Version	| 2.0.0
SPL Version	| n/a

### Describe the problem you're observing

configure fails in the sed test with the following error:
```
checking for sed --in-place... mktemp: (null): Invalid argument     
../configure: line 16614: can't create : nonexistent directory
configure: error: /bin/sed does not support in-place
```
### Describe how to reproduce the problem

backup mktemp binary and create a `mktemp` symlink pointing to busybox.
run configure.

### Include any warning/errors/backtraces from the system logs

The problem appears to be that busybox configure requires 6 trainling X'es while GNU coreutils only requires 3.
https://github.com/openzfs/zfs/blob/dcbf8474933cb22b90dbd7f514c8a4c71ea4c1cf/config/always-sed.m4#L7

**Comment by ncopa** [comment]
This fixes it:
```diff
diff --git a/config/always-sed.m4 b/config/always-sed.m4
index 19633e1..3d7ae28 100644
--- a/config/always-sed.m4
+++ b/config/always-sed.m4
@@ -4,7 +4,7 @@ dnl #
 AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_SED], [
        AC_REQUIRE([AC_PROG_SED])dnl
        AC_CACHE_CHECK([for sed --in-place], [ac_cv_inplace], [
-               tmpfile=$(mktemp conftest.XXX)
+               tmpfile=$(mktemp conftest.XXXXXX)
                echo foo >$tmpfile
                AS_IF([$SED --in-place 's#foo#bar#' $tmpfile 2>/dev/null],
                      [ac_cv_inplace="--in-place"],
```

**Comment by behlendorf** [comment]
Resolved by #11269.
